contracts/colony/ColonyDomains.sol

```diff
@@ -124,6 +127,13 @@ contract ColonyDomains is ColonyStorage {
     return domainCount;
   }
 
+  function setDomainReputationScaling(uint256 _domainId, uint256 _factor) public stoppable auth {
```

Comment:
We have a big problem with this implementation, as recovery mode can be used to set a domain's skill to an arbitrary skill id, allow colonies to set each-other's scaling, which is obviously unacceptable.

Reply:
I don't love storing the scaling factor on the colony, since it involves making external calls as we walk up the skills tree. Hard to defend from a gas perspective. Can't we add the domain skill as a protected slot in recovery mode? We already have infrastructure for that.

Reply:
No, because it's not currently a protected slot (so a colony could set it, then upgrade)

Reply:
Mm, that's frustrating. Your optimization is better, but I do worry about the gas costs of earning reputation being significantly increased. Doesn't seem like there's a better way though.

Is there any possibility of protecting the slot in the next version, and then updating the functionality at some point down the road, once all colonies have upgraded? It does seem like that would be worth it long-term, given how often this function will be called. Definitely a high-value optimization.

Reply:
If it helps, the limit still seems to be about 150 depth which was the limit when the storage was on colonyNetwork in the original implementation.

> Is there any possibility of protecting the slot in the next version, and then updating the functionality at some point down the road, once all colonies have upgraded? It does seem like that would be worth it long-term, given how often this function will be called. Definitely a high-value optimization.

In principle, yes (if we also prevented deployment of earlier versions, which we currently allow), but I'm fairly confident in saying that we would never get to a point where all colonies had upgraded.

Reply:
So it goes, I guess.

Reply:
I mean, that's going to be true with any implementation of this feature, as defined, as wherever the data is stored, you have to work your way up the tree to calculate the scaling factor. The question is by how much. This implementation is slightly worse than the original one, but really not by much.

I haven't worked it out, but it's possible that given how memory costs work (i.e. quadratically, if you start requesting a lot), it might be more gas efficient to 'page' the allParents function and call it a couple of times, but it doesn't look like that's needed for now.